digdag-core/src/main/java/io/digdag/core/workflow/AttemptBuilder.java

```diff
@@ -110,14 +111,26 @@ public AttemptRequest buildFromStoredWorkflow(
         return monitors.build();
     }
 
-    private Config buildSessionParameters(Config overrideParams, Optional<Scheduler> sr, Instant sessionTime, ZoneId timeZone)
+    private Config buildSessionParameters(Config overrideParams, Optional<Scheduler> sr, Instant sessionTime, ZoneId timeZone, Optional<Instant> lastExecutedSessionTime)
     {
         Config params = overrideParams.deepCopy();
         if (sr.isPresent()) {
             Instant lastSessionTime = sr.get().lastScheduleTime(sessionTime).getTime();
             Instant nextSessionTime = sr.get().nextScheduleTime(sessionTime).getTime();
             params.set("last_session_time", formatSessionTime(lastSessionTime, timeZone));
             params.set("next_session_time", formatSessionTime(nextSessionTime, timeZone));
+            if (lastExecutedSessionTime.isPresent()) {
+                // TODO: Come up with a better names for these parameters?
+
+                // This is the last session that was processed. Normally this will be the immediately preceding session,
+                // but it can be an earlier session if some were skipped due to skip_on_overtime = true.
+                params.set("last_processed_session_time", formatSessionTime(lastExecutedSessionTime.get(), timeZone));
+
+                // This is the first session time slot that should be included if the user wants to process any preceding skipped sessions.
+                // Normally this is the same as the current session_time but can be an earlier session if some were skipped.
+                Instant firstUnprocessedSessionTime = sr.get().nextScheduleTime(lastExecutedSessionTime.get()).getTime();
+                params.set("first_unprocessed_session_time", formatSessionTime(firstUnprocessedSessionTime, timeZone));
```

Comment:
What would the value of `last_successful_session_time` be for the first scheduled session? 

My thinking was that `first_unprocessed_session_time` would be equal to e.g. `start_time` for the first scheduled session. However, `last_successful_session_time` would not be set because there would not be a preceding "successful" session. This would make `last_successful_session_time` unusable in workflows that have a "first" scheduled execution.

What was the value of `:last_successful_timestamp` in the first scheduled query in "Metric Insights" ?